### PR TITLE
Handle multiple CNAME resolve and debug log http redirect

### PR DESCRIPTION
**Pull request by edwardrf**
For https://github.com/DefangLabs/defang-mvp/issues/981

### CODE REVIEW

**Comment by jordanstephens** [comment]
In what context does this happen?

**Review by lionello — APPROVED** [review]

File: src/pkg/dns/resolver_integration_test.go [hunk 1]
```diff
@@ -43,4 +43,15 @@ func TestRootResolver(t *testing.T) {
 			t.Errorf("Expected same IP addresses, got %v <> %v", ips, nIPs)
 		}
 	})
+
+	t.Run("LookupIPAddr follows multiple CNAME redirects", func(t *testing.T) {
+		r := RootResolver{}
+		ips, err := r.LookupIPAddr(context.Background(), "cname-a.gnafed.click")
```
Comment (lionello) [inline src/pkg/dns/resolver_integration_test.go:49]: add to Pulumi

**Review by lionello — APPROVED** [review]

**Comment by edwardrf** [comment]
> In what context does this happen?

This happens when we try generate cert for the user, we need to check if the user has setup their DNS correctly, which,if the user has a local DNS cache, could take a long time. Thus we created this tool to query directly against the authoritative server.

There might be a case the user has a more complicated DNS setup where multiple CNAMES are used, which is still valid.

File: src/pkg/dns/resolver_integration_test.go (hunk 1 above)
Comment (edwardrf) [reply src/pkg/dns/resolver_integration_test.go:49]: Added: https://github.com/DefangLabs/defang-mvp/pull/995

**Pull request merged**